### PR TITLE
Make the code more modern and readable

### PR DESCRIPTION
**Pull request by ninofiliu**
One of the painpoints I had when learning about PWAs by following [the MDN guides on the subject](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps) that heavily reference these example apps was the usage of old syntax that oftentimes made the code very hard to read, like the following:

```js
self.addEventListener('fetch', function(e) {
  e.respondWith(
    caches.match(e.request).then(function(r) {
      console.log('[Service Worker] Fetching resource: '+e.request.url);
      return r || fetch(e.request).then(function(response) {
        return caches.open(cacheName).then(function(cache) {
          console.log('[Service Worker] Caching new resource: ' + e.request.url);
          cache.put(e.request, response.clone());
          return response;
        });
      });
    })
  );
});
```

It took me a while at first to understand what was actually awaited in the `respondWith` :sweat_smile: 

By following the *"fix it or shut up"* philosophy, I am, in this PR:

- Linting the JS files: there was some uneven indentation, some variables where not used, there was tabs instead of spaces, etc. I used [airbnb's eslint config](https://www.npmjs.com/package/eslint-config-airbnb) to do the bulk of the work automatically
- Using async/await and IIFEs when it makes the code more readable

This is my first PR here, don't hesitate to tell me if I'm doing something wrong!

### CODE REVIEW

**Review by chrisdavidmills — APPROVED** [review]
This is a fantastic set of improvements, thank you @ninofiliu !

And apologies for taking so long to review this.

File: js13kpwa/sw.js [hunk 1]
```diff
@@ -20,37 +20,34 @@ var appShellFiles = [
...
-          cache.put(e.request, response.clone());
-          return response;
-        });
-      });
-    })
-  );
+self.addEventListener('fetch', (e) => {
+  e.respondWith((async () => {
+    const r = await caches.match(e.request);
+    console.log(`[Service Worker] Fetching resource: ${e.request.url}`);
```
Comment (waet999999) [inline js13kpwa/sw.js:45]: e

Comment (waet999999) [inline js13kpwa/sw.js:45]: E

Comment (waet999999) [inline js13kpwa/sw.js:45]: E

Comment (waet999999) [inline js13kpwa/sw.js:45]: e

**Pull request merged**